ntp-proto/src/packet/crypto.rs

```diff
@@ -205,3 +262,46 @@ impl std::fmt::Debug for AesSivCmac512 {
         f.debug_struct("AesSivCmac512").finish()
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_aes_siv_cmac_256() {
+        let mut testvec: Vec<u8> = (0..16).collect();
+        testvec.resize(testvec.len() + 32, 0);
+        let key = AesSivCmac256::new([0u8; 32].into());
+        let EncryptResult {
+            nonce_length,
+            ciphertext_length,
+        } = key.encrypt(&mut testvec, 16, &[]).unwrap();
+        let result = key
+            .decrypt(
+                &testvec[..nonce_length],
+                &testvec[nonce_length..(nonce_length + ciphertext_length)],
+                &[],
```

Comment:
should we also test with associated data at this point? or else just remove it from the `Cipher` trait if we don't use it.